src/data/list/defs.lean

```diff
@@ -936,4 +935,27 @@ Asynchronous version of `list.map`.
 meta def map_async_chunked {α β} (f : α → β) (xs : list α) (chunk_size := 1024) : list β :=
 ((xs.to_chunks chunk_size).map (λ xs, task.delay (λ _, list.map f xs))).bind task.get
 
+/-!
+  We add some n-ary versions of `list.zip_with` for functions with more than two arguments.
+  These can be written in terms of `list.zip` and `list.map`, but these are more convenient
+  (and efficient) to use.
+  For example, `zip_with3 f xs ys zs` could also be written as
+  `(zip xs $ zip ys zs).map $ λ ⟨x, y, z⟩, f x y z`.
```

Comment:
I recall doc-gen will interpret these spaces as a monospaced text block.
```suggestion
We add some n-ary versions of `list.zip_with` for functions with more than two arguments.
These can be written in terms of `list.zip` and `list.map`, but these are more convenient
(and efficient) to use.
For example, `zip_with3 f xs ys zs` could also be written as
`(zip xs $ zip ys zs).map $ λ ⟨x, y, z⟩, f x y z`.
```

Reply:
Didn't we find a better way to write these on zulip?

Reply:
You're right. I forgot to incorporate his suggestion. I think I still prefer the recursive definition for the actual definition, but I added it as a comment.